synapse/events/snapshot.py

```diff
@@ -254,6 +298,131 @@ async def get_prev_state_ids(
         )
 
 
+@attr.s(slots=True, auto_attribs=True)
+class UnpersistedEventContext(UnpersistedEventContextBase):
+    """
+    The event context holds information about the state groups for an event. It is important
+    to remember that an event technically has two state groups: the state group before the
+    event, and the state group after the event. If the event is not a state event, the state
+    group will not change (ie the state group before the event will be the same as the state
+    group after the event), but if it is a state event the state group before the event
+    will differ from the state group after the event.
+    This is a version of an EventContext before the new state group (if any) has been
+    computed and stored. It contains information about the state before the event (which
+    also may be the information after the event, if the event is not a state event). The
+    UnpersistedEventContext must be converted into an EventContext by calling the method
+    'persist' on it before it is suitable to be sent to the DB for processing.
+
+        state_group_after_event:
+             The state group after the event. This will always be None until it is persisted.
+             If the event is not a state event, this will be the same as
+             state_group_before_event.
+
+        state_group_before_event:
+            The ID of the state group representing the state of the room before this event.
+
+        state_delta_due_to_event:
+            If the event is a state event, then this is the delta of the state between
+             `state_group` and `state_group_before_event`
+
+        prev_group_for_state_group_after_event:
+            If it is known, ``state_group_after_event``'s previous state group.
+
+             If the event is a state event, this is normally the same as
+             ``state_group_before_event``.
+
+        delta_ids_to_state_group_after_event:
+             If ``prev_group_for_state_group_after_event`` is not None, the state delta
+             between ``prev_group_for_state_group_after_event`` and ``state_group_after_event``.
```

Comment:
These docstrings need to be updated (its annoying that python makes us have to keep them separate from the definition!)

Reply:
Ah yeah I totally forgot about those, have fixed.